ast/term.go

```diff
@@ -1411,6 +1420,14 @@ func (s *set) String() string {
 	return b.String()
 }
 
+func (s *set) sortedKeys() []*Term {
+	if s.numInserts > 0 {
+		sort.Sort(s.keys)
```

Comment:
[nit] We could have done this here, and not introduce the interface above, couldn't we?

```suggestion
		sort.Sort(termSlice(s.keys))
```

Reply:
That is a solid suggestion, and I'm rolling that in shortly. :smiley: